misc/extension_api_validation/4.2-stable.expected

```diff
@@ -14,3 +14,27 @@ Validate extension JSON: Error: Field 'classes/TileData/methods/get_navigation_p
 Validate extension JSON: Error: Field 'classes/TileData/methods/get_occluder/arguments': size changed value in new API, from 1 to 4.
 
 Added optional argument to get_navigation_polygon and get_occluder to specify a polygon transform.
+
+GH-84976
+--------
+Validate extension JSON: Error: Field 'classes/RenderingDevice/enums/FinalAction/values/FINAL_ACTION_CONTINUE': value changed value in new API, from 2.0 to 0.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/enums/FinalAction/values/FINAL_ACTION_MAX': value changed value in new API, from 3.0 to 2.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/enums/InitialAction/values/INITIAL_ACTION_CLEAR': value changed value in new API, from 0.0 to 1.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/enums/InitialAction/values/INITIAL_ACTION_CLEAR_REGION_CONTINUE': value changed value in new API, from 2.0 to 0.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/enums/InitialAction/values/INITIAL_ACTION_CONTINUE': value changed value in new API, from 5.0 to 0.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/enums/InitialAction/values/INITIAL_ACTION_DROP': value changed value in new API, from 4.0 to 2.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/enums/InitialAction/values/INITIAL_ACTION_KEEP': value changed value in new API, from 3.0 to 0.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/enums/InitialAction/values/INITIAL_ACTION_MAX': value changed value in new API, from 6.0 to 3.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/methods/buffer_clear/arguments': size changed value in new API, from 4 to 3.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/methods/buffer_update/arguments': size changed value in new API, from 5 to 4.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/methods/draw_list_begin/arguments': size changed value in new API, from 10 to 9.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/methods/texture_clear/arguments': size changed value in new API, from 7 to 6.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/methods/texture_copy/arguments': size changed value in new API, from 10 to 9.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/methods/texture_resolve_multisample/arguments': size changed value in new API, from 3 to 2.
+Validate extension JSON: Error: Field 'classes/RenderingDevice/methods/texture_update/arguments': size changed value in new API, from 4 to 3.
+Validate extension JSON: Error: Hash changed for 'classes/RenderingDevice/methods/compute_list_end', from E9B4FA8E to 19365687. This means that the function has changed and no compatibility function was provided.
+Validate extension JSON: Error: Hash changed for 'classes/RenderingDevice/methods/draw_list_end', from E9B4FA8E to 19365687. This means that the function has changed and no compatibility function was provided.
```

Comment:
These two shouldn't be listed as expected - instead, compatibility methods should be added which call the method with the new signature.

Reply:
I think part of the problem is these two were changed with a previous compatibility method in a previous PR, but there's no way to indicate they've been changed again?

https://github.com/DarioSamo/godot/blob/f23b0fdf62cdd0ffa27298531027978e29dcb4b9/servers/rendering/rendering_device.compat.inc#L110-L111

Reply:
Oh, you can't like "chain" compatibility methods together, you need to have a compatibility method for each version of the signature. So, for example, the old compatibility method was for when the default value of the argument on `draw_list_end()` was `7`. But the current version of the method has a default value of `0x7FFF` so you need to add a new compatibility method with that as the default value, for example:

```
ClassDB::bind_compatibility_method(D_METHOD("draw_list_end", "post_barrier"), &RenderingDevice::_draw_list_end_bind_compat_84976, DEFVAL(0x7FFF));
```

Hopefully that makes sense!

Reply:
Thanks, just added those.